tools/pythonpkg/src/python_udf.cpp

```diff
@@ -43,9 +43,34 @@ static py::object ConvertDataChunkToPyArrowTable(DataChunk &input, const ClientP
 	return pyarrow::ToArrowTable(types, names, ConvertToSingleBatch(types, names, input, options), options);
 }
 
+// If these types are arrow canonical extensions, we must check if they are registered.
+// If not, we should error.
+void AreExtensionsRegistered(const LogicalType &arrow_type, const LogicalType &duckdb_type) {
+	if (arrow_type != duckdb_type) {
+		// Is it a UUID Registration?
+		if (arrow_type.id() == LogicalTypeId::BLOB && duckdb_type.id() == LogicalTypeId::UUID) {
+			throw InvalidConfigurationException(
+			    "Mismatch on return type from Arrow object (%s) and DuckDB (%s). It seems that you are using the UUID "
+			    "arrow canonical extension, but the same is not yet registered. Make sure to register it first with "
+			    "e.g., pa.register_extension_type(UUIDType()). ",
+			    arrow_type.ToString(), duckdb_type.ToString());
+		}
+		// Is it a JSON Registration
+		if (!arrow_type.IsJSONType() && duckdb_type.IsJSONType()) {
+			throw InvalidConfigurationException(
+			    "Mismatch on return type from Arrow object (%s) and DuckDB (%s). It seems that you are using the JSON "
+			    "arrow canonical extension, but the same is not yet registered. Make sure to register it first with "
+			    "e.g., pa.register_extension_type(JSONType()). ",
+			    arrow_type.ToString(), duckdb_type.ToString());
+		}
+	}
+}
 static void ConvertArrowTableToVector(const py::object &table, Vector &out, ClientContext &context, idx_t count) {
 	// Create the stream factory from the Table object
-	auto stream_factory = make_uniq<PythonTableArrowArrayStreamFactory>(table.ptr(), context.GetClientProperties());
+	auto ptr = table.ptr();
+	py::gil_scoped_release gil;
```

Comment:
I'm not sure this is safe

The scope of this function is quite big, we're both binding and executing the arrow scan and passing it a `PyObject*`, this has to be interacting with python at least once in this process

Reply:
I think the danger is mostly in `PythonTableArrowArrayStreamFactory::Produce`

But that seems to already grab the GIL, so it might be fine??

Reply:
Can you pinpoint exactly where this interacts with the arrow object outside its c data interface?

I released this because the internal function of the scan of the arrow object requires the gil itself, hence deadlocking.